pipelines/scipipe/ap_verify.groovy

```diff
@@ -56,6 +56,16 @@ notify.wrap {
     // butler
     def runSlug = "${datasetSlug(conf)}^${codeSlug(conf)}"
 
+    // these configs are used to construct a shell command line; guard against code injection
```

Comment:
I think anyone who can commit to jenkins-dm-jobs can convince it to do anything they want, but good for you implementing this anyway.

Reply:
As you say, there's no way to harden the script against direct edits, but I'm uncomfortable with code that assumes all input validation has already been done upstream. This way, if there is (or will be) another way to define pipeline YAMLs, the script will still be valid.